meteor/client/ui/Shelf/BucketPanel.tsx

```diff
@@ -919,3 +944,81 @@ export const BucketPanel = translateWithTracker<Translated<IBucketPanelProps>, I
 		)
 	)
 )
+
+interface DropzoneHolderProps {
+	id: string
+	bucketId: BucketId
+	url: string
+	hidden: boolean
+	t: TFunction<'translation', undefined>
+	showStyleBaseId: ShowStyleBaseId
+}
+const DropzoneHolder = (props: DropzoneHolderProps) => {
+	let frame: HTMLIFrameElement | null = null
+	let mounted = false
+
+	const setDropzoneElement = (frameRef: HTMLIFrameElement | null) => {
+		frame = frameRef
+		if (frame && !mounted) {
+			registerHandlers()
+			mounted = true
+		} else {
+			unregisterHandlers()
+			mounted = false
+		}
+	}
+	const registerHandlers = () => {
+		window.addEventListener('message', onMessage)
+	}
+	const unregisterHandlers = () => {
+		window.removeEventListener('message', onMessage)
+	}
+
+	const onMessage = (event: MessageEvent) => {
+		if (event.source === frame?.contentWindow) {
+			if (event.data && event.data.event === 'drop') {
+				RundownViewEventBus.emit(RundownViewEvents.ITEM_DROPPED, {
+					id: props.id,
+					bucketId: props.bucketId,
+					ev: event,
+				})
+			}
+
+			if (event.data && event.data.event === 'data' && event.data.data.trim().endsWith('</mos>')) {
+				RundownViewEventBus.emit(RundownViewEvents.ITEM_DROPPED, {
+					id: props.id,
+					bucketId: props.bucketId,
+					message: event.data.data,
+					ev: event,
+				})
+			}
+
+			if (event.data && event.data.event === 'error') {
+				RundownViewEventBus.emit(RundownViewEvents.ITEM_DROPPED, {
+					id: props.id,
+					bucketId: props.bucketId,
+					error: event.data.message,
+					ev: event,
+				})
+			}
+		}
+	}
+
+	React.useEffect(
+		() => () => {
+			unregisterHandlers()
+		},
+		[]
+	)
```

Comment:
I would suggest re-arranging this to a pattern where the register- and unregisterHandlers are joined in a single hook, something like: 

```suggestion
	const [dropzoneElementRef, setDropzoneElementRef] = React.useState<HTMLElement>(null)
	const mounted = !!dropzoneElementRef
	
	React.useEffect(
		() => {
			if (!dropzoneElementRef) return
			
			const registerHandlers = () => {
				window.addEventListener('message', onMessage)
			}
			const unregisterHandlers = () => {
				window.removeEventListener('message', onMessage)
			}
			
			registerHandlers()
			
			return () => {
				unregisterHandlers()
			}
		},
		[dropzoneElementRef]
	)
```